File: frontend/src/services/mentorService.js
import api from './api';

const mentorService = {
  getDashboard: async () => {
    const response = await api.get('/api/mentor/dashboard');
    return response.data;
  },

  getStudents: async (page = 0, size = 10) => {
    const response = await api.get(`/api/mentor/students?page=${page}&size=${size}`);
    return response.data;
  },

  getStudentDetails: async (studentId) => {
    const response = await api.get(`/api/mentor/students/${studentId}`);
    return response.data;
  },

  getStudentProgress: async (studentId) => {
    const response = await api.get(`/api/mentor/students/${studentId}/progress`);
    return response.data;
  },

  getStudentActivity: async (studentId) => {
    const response = await api.get(`/api/mentor/students/${studentId}/activity`);
    return response.data;
  },

  getStudentAnalytics: async (studentId, range = '7d') => {
    const response = await api.get(`/api/mentor/students/${studentId}/analytics?range=${range}`);
    return response.data;
  },

  exportStudentsCsv: async () => {
    const response = await api.get('/api/mentor/export/students.csv', { responseType: 'blob' });
    const url = window.URL.createObjectURL(new Blob([response.data]));
    const link = document.createElement('a');
    link.href = url;
    link.setAttribute('download', 'students.csv');
    document.body.appendChild(link);
    link.click();
    link.remove();
  }
};

export default mentorService;

File: backend/src/main/java/com/example/studentdashboard/service/MentorService.java
package com.example.studentdashboard.service;

import com.example.studentdashboard.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.io.ByteArrayInputStream;
import java.util.List;

public interface MentorService {
    MentorDashboardDto getDashboard(Long mentorId);
    Page<MentorStudentSummaryDto> getStudents(Long mentorId, Pageable pageable);
    StudentDetailsDto getStudentDetails(Long mentorId, Long studentId);
    List<LessonProgressDto> getStudentProgress(Long mentorId, Long studentId);
    List<ActivityEventDto> getStudentActivity(Long mentorId, Long studentId);
    LearningTrendDto getStudentAnalytics(Long mentorId, Long studentId, String range);
    ByteArrayInputStream exportStudentsCsv(Long mentorId);
}
